### PR TITLE
For code review

### PR DESCRIPTION
**Pull request by orestis-kaplan**

### CODE REVIEW

File: boxes.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/*jshint esversion: 6 */
+const gameBoard = (function(){
+  //Private fields
+  let numberOfBoxes = 9;
+  let boxes= document.createElement('div');
```
Comment (rvvergara) [inline boxes.js]: Let us separate DOM related logic from our factory logic :)

**Review by rvvergara — CHANGES_REQUESTED** [review]
Hi @userman95,

Nice User Interface!!! :) 

### Architecture change

We need at least 2 factories. Using `class` is ok, though:

1. `player` factory
2. `board` factory

Currently what we have is only a `player` class/factory. The `board` (as contained in `boxes.js` file) is a factory but it combines a lot of DOM related stuff. If you remember the OOP concepts from the Ruby TicTactoe, those same concepts apply here. We want our objects to be independent of the outside world and just communicate amongst each other using public methods :) 

Here are some suggestions on how we can architect the player and board factories:

### Player Factory:

**Data contained** -> `player` instances/objects should have a `name`, `symbol` and an array that stores all the `moves` for that particular player (contains indices of the squares a player has chosen).
**Actions/functions/methods** -> contains only one method which we can call a `turn` or `move` method. This method will take an index and the `board` as arguments and:

```javascript
const turn => (index, board) => {
  moves.push(index);
  board.updateSquare(index, symbol);
}
```
In short, the `moves` method will represent a player making a "move". The index of course may come from different sources. In this case it will come from the user clicking a particular square in the board's display.

### Board factory

For board this is my suggested architecture:

```javascript
const Board = () => {
  const boxes = [0, 1, 2 ... 8]
  const winningCombinations = [[0,1, 2]... // all the index combinations that will make a player win];
 
  const isFull = () => // method that checks if all elements in "boxes" are either "X" or "O" => returns only true or false

const checkWin = (player) => // method that checks a player's moves array and compares it with the winningCombinations array to see if a player has achieved any of the winning combo

const updateSquare = (index, symbol) => // a method that updates a particular square in "boxes" to 
the player's symbol (as you'll remember we call this in player's move method)
}
```

The above is a suggestion on how to make your player and board objects reusable and decoupled from the DOM and from each other :)

**Review by inhaq — CHANGES_REQUESTED** [review]
Good job on updates. Although the functionality is not working as expected. When I press each box for 9 times, it ends up in a draw. Functionality should be that it should avoid clicking again on the same box.
Kindly update and resubmit.

**Review by inhaq — APPROVED** [review]
Awesome, It's 👍🏻

**Pull request merged**